### DISCUSSION TITLE
Swagger docs break sporadically with a 404

### DISCUSSION THREAD

**Discussion by ahmed-wajid** [fastapi/fastapi#8410] [Questions]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [ ] I commit to help with one of those options 👆

### Example Code

```python
app = FastAPI(
        title="Foobar Framework",
        root_path="/foobar",
    )
```

### Description

When we deploy the application behind a reverse proxy such as an nginx Ingress Controller, and add a root path, within an AKS cluster, the worker is initially killed with the error `[1] [WARNING] Worker with pid 12 was terminated due to signal 9`. But the worker later does start, and the API is working. However, the new worker, no longer has `"servers":[{"url":"/foobar"}]` in the `openapi.json` and hence the docs no longer work and give a 404. The API itself, and the business logic just work absolutely fine. 

To reproduce the issue, we run the API locally behind an nginx container with the following conf:
```
server {
  listen 8080;

  location /foobar/ {
    rewrite ^/foobar(.*) $1 break;
    proxy_pass http://foobar-backend:8991;
  }
}
```

The API is launched with: 
`python -m gunicorn api:app --workers=1 --threads=4 -k uvicorn.workers.UvicornWorker --bind 0.0.0.0:8991`

And then, kill the process manually in the container. It sporadically loses the Server URL context and docs are broke. We assume that for some reason, the `root_path` is not picked up by the worker when it respawns. 

### Operating System

Linux

### Operating System Details

Running in Ubuntu based docker image 

### FastAPI Version

0.79.1

### Python Version

Python 3.9.15

### Additional Context

if we skip the auto generation of the root path in servers, this workaround does the job:

```
    app = FastAPI(
        title="Foobar Framework",
        root_path="/foobar",
        servers=[
            {"url": "http://localhost:8080/foobar", "description": "Local Dev environment"},
        ],
        root_path_in_servers=False,
    )
```

**Comment by ahmed-wajid** [comment]
Perhaps this is due to `openapi_prefix` and `root_path`, because if I set them both, so far it seems to be working fine.

**Comment by iudeen** [comment]
If you're serving behind a proxy, you need to specify that proxied path in root_path.

**Comment by ahmed-wajid** [comment]
Thank you for your reply. 

As mentioned, we do specify the proxied path, and it does work. It only breaks sporadically the docs, when the subprocess reloads. This happens less frequently when I specified both `openapi_prefix` and `root_path`. I do acknowledge that `openapi_prefix`is deprecated.

**Comment by iudeen** [comment]
Uhmm weird. Anything on logs?

**Comment by ahmed-wajid** [comment]
Unfortunately, nothing in the logs. Even with trace enabled.
